test/unit/Cardano/DBLayer/MVarSpec.hs

```diff
@@ -1,14 +1,139 @@
+{-# LANGUAGE FlexibleInstances #-}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
 module Cardano.DBLayer.MVarSpec
     ( spec
     ) where
 
 import Prelude
 
+import Cardano.DBLayer
+    ( DBLayer (..), PrimaryKey (..) )
 import Cardano.DBLayer.MVar
-    ()
+    ( newDBLayer )
+import Cardano.Wallet
+    ( Wallet (..), WalletId (..), initWallet )
+import Cardano.Wallet.Primitive
+    ( IsOurs (..) )
+import Control.Concurrent.Async
+    ( mapConcurrently_ )
+import Control.DeepSeq
+    ( NFData )
+import Control.Monad.IO.Class
+    ( liftIO )
+import Data.List.NonEmpty
+    ( NonEmpty ((:|)) )
 import Test.Hspec
-    ( Spec )
+    ( Spec, describe, it, shouldBe )
+import Test.QuickCheck
+    ( Arbitrary (..), Property, checkCoverage, choose, vectorOf )
+import Test.QuickCheck.Monadic
+    ( monadicIO )
 
+import qualified Data.Set as Set
+import qualified Data.Text as T
 
 spec :: Spec
-spec = return ()
+spec = do
+    describe "DB works as expected" $ do
+        it "readCheckpoints works properly"
```

Comment:
Minor suggestion in the title here but, maybe we give some insights about what "works properly" means, and states the actual properties being checked like "readCheckpoints . putCheckpoints yields inserted checkpoints"

Reply:
:+1: